<a href="https://colab.research.google.com/github/AnaSantillanal246102/DataScience/blob/main/Analisis_Datos/S1_Analisis_de_Datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Ejemplo 01: Datos estructurados y estimados de locación

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Importar librerías
import pandas as pd

# Cargar el dataset
df = pd.read_csv('/content/drive/MyDrive/DataScienceBedu/Datasets/S01/Ejemplo_01_Titanic_Dataset.csv') # Modifica la ruta de acuerdo a tu entorno de trabajo

# Mostrar las primeras filas del dataset
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,0,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,1,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,0,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,0,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,1,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


In [ ]:
# Calcular la media
promedio_edad = df['Age'].mean()
print(f'El promedio de edad de los pasajeros en el Titanic fue: {promedio_edad}')

# Calcular la mediana
mediana_edad = df['Age'].median()
print(f'El valor central de la edad de los pasajeros en el Titanic fue: {mediana_edad}')

# Calcular la moda
moda_edad = df['Age'].mode()[0]
print(f'La edad más común de los pasajeros en el Titanic fue: {moda_edad}')

El promedio de edad de los pasajeros en el Titanic fue: 30.272590361445783
El valor central de la edad de los pasajeros en el Titanic fue: 27.0
La edad más común de los pasajeros en el Titanic fue: 21.0


Ejemplo 02: Valores atípicos y estimados de variabilidad

In [ ]:
# Importar librerías
import pandas as pd

# Cargar el dataset
df = pd.read_csv('/content/drive/MyDrive/DataScienceBedu/Datasets/S01/Ejemplo_01_Titanic_Dataset.csv') # Modifica la ruta de acuerdo a tu entorno de trabajo

# Mostrar las primeras filas del dataset
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,0,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,1,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,0,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,0,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,1,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


In [ ]:
# Calcular la varianza de la tarifa
varianza_fare = df['Fare'].var()
print(f'La varianza de la tarifa de los pasajeros en el Titanic es: {varianza_fare:,.2f}')

# Nota: Este valor representa el valor de tarifa elevado al cuadrado, por lo que no tiene unidades monetarias.

La varianza de la tarifa de los pasajeros en el Titanic es: 3,125.66


In [ ]:
# Calcular la desviación estándar de la tarifa
desviacion_estandar_fare = df['Fare'].std()
print(f'La desviación estándar de la tarifa de los pasajeros en el Titanic es: ${desviacion_estandar_fare:,.2f}')

La desviación estándar de la tarifa de los pasajeros en el Titanic es: $55.91


In [ ]:
# Calcular el rango de la edad
edad_max = df['Age'].max()
edad_min = df['Age'].min()
rango_edad = edad_max - edad_min
print(f'El rango de la edad de los pasajeros en el Titanic es: {rango_edad}')

El rango de la edad de los pasajeros en el Titanic es: 75.83


In [ ]:
from scipy import stats
import pandas as pd

# Eliminar valores NaN en la columna 'Age'
edad_sin_nan = df['Age'].dropna()

# Calcular la media truncada
media_truncada_edad = stats.trim_mean(edad_sin_nan, 0.05)
print(f'La media truncada de la edad de los pasajeros en el Titanic es: {media_truncada_edad:.2f}')

La media truncada de la edad de los pasajeros en el Titanic es: 30.02


Ejemplo 03: Cuartiles y Percentiles

In [ ]:
# Importar librerías
import pandas as pd

# Cargar el dataset
df = pd.read_csv('/content/drive/MyDrive/DataScienceBedu/Datasets/S01/Ejemplo_01_Titanic_Dataset.csv') # Modifica la ruta de acuerdo a tu entorno de trabajo

# Calcular los cuartiles de la tarifa
Q1_fare = df['Fare'].quantile(0.25)
Q2_fare = df['Fare'].quantile(0.50)
Q3_fare = df['Fare'].quantile(0.75)

print(f'Primer cuartil (Q1) de la tarifa: ${Q1_fare:.2f}')
print(f'Segundo cuartil (Q2) de la tarifa (mediana): ${Q2_fare:.2f}')
print(f'Tercer cuartil (Q3) de la tarifa: ${Q3_fare:.2f}')

Primer cuartil (Q1) de la tarifa: $7.90
Segundo cuartil (Q2) de la tarifa (mediana): $14.45
Tercer cuartil (Q3) de la tarifa: $31.50


In [ ]:
# Eliminar valores NaN en la columna 'Age'
edad_sin_nan = df['Age'].dropna()

# Calcular el percentil 90 de la edad
percentil_90_edad = edad_sin_nan.quantile(0.90)

print(f'El percentil 90 de la edad de los pasajeros en el Titanic es: {percentil_90_edad:.2f} años')

El percentil 90 de la edad de los pasajeros en el Titanic es: 50.00 años


#Reto 1
Instrucciones
📂 Carga el dataset: Descarga el archivo Reto_01_Laptops_Dataset.csv y súbelo a tu Google Drive. Luego, carga el dataset en un DataFrame de Pandas.

📊 Calcular los estimados de locación: Calcula la media, mediana y moda de la columna price del dataset.

🔍 Análisis de resultados: Compara los resultados obtenidos e interpreta qué información nos proporcionan sobre la distribución de los precios de las laptops en el dataset.

✅ Desafío adicional: Si deseas practicar un poco más, puedes calcular lo mismo para la columna screen_size del dataset. ¿Qué información adicional puedes obtener de estos estimados de locación?

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/DataScienceBedu/Datasets/S01/Reto_01_Laptops_Dataset.csv')
# Calcular la media
promedio_precio = df['price'].mean()
print(f'El promedio del precio de las laptops fue: ${promedio_precio:,.2f}')
# Calcular la mediana
mediana_precio = df['price'].median()
print(f'El valor central del precio de las laptops es: ${mediana_precio:,.2f}')
# Calcular la moda
moda_precio = df['price'].mode()[0]
print(f'El precio más común de las laptops es: ${moda_precio:,.2f}')

El promedio del precio de las laptops fue: $84,889.25
El valor central del precio de las laptops es: $64,690.00
El precio más común de las laptops es: $59,990.00


Este analísis nos indica que la media se inclina a la derecha, sumado a que la moda predomina hacia la izquierda del valor central, nos indica que hay outliers muy distantes de la media, porque la moda se encuentra a la izquierda entonces sabemos que no se tratan de muchos datos que afecten del lado derecho.

In [ ]:
# Calcular la media
promedio_size_screen = df['screen_size'].mean()
print(f'El promedio del tamaño de las pantallas de las laptops fue: {promedio_size_screen}')
# Calcular la mediana
mediana_size_screen = df['screen_size'].median()
print(f'El valor central del tamaño de las pantallas de las laptops es: {mediana_size_screen}')
# Calcular la moda
moda_size_screen = df['screen_size'].mode()[0]
print(f'El tamaño de las pantallas más común de las laptops es: {moda_size_screen}')

El promedio del tamaño de las pantallas de las laptops fue: 15.219007021063188
El valor central del tamaño de las pantallas de las laptops es: 15.6
El tamaño de las pantallas más común de las laptops es: 15.6


En cambio este analisis, nos indica que el promedio no es distante del valor central y es reforzado porque la moda compagina con el valor central.

#Reto 2
Instrucciones
📂 Carga el dataset: Descarga el archivo Salary_Professions.csv y súbelo a tu Google Drive. Luego, carga el dataset en un DataFrame de Pandas.

📊 Identificar valores atípicos: Utiliza métodos estadísticos para identificar valores atípicos en la columna salary.

📐 Calcular los estimados de variabilidad:

Calcula la varianza de la columna salary.
Calcula la desviación estándar de la columna salary.
Calcula el rango de la columna salary.
🔍 Análisis de resultados: Compara los resultados obtenidos e interpreta qué información nos proporcionan sobre la distribución de los salarios en el dataset.

✅ Desafío adicional: Si deseas practicar un poco más, puedes calcular la media truncada para quitar los valores atípicos y compararla con la media original. ¿Qué diferencias observas entre ambos valores?

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/DataScienceBedu/Datasets/S01/Reto_02_Salary_Professions.csv')
varianza_salary = df['SALARY'].var()
print(f'La varianza de los salarios entre profesiones es: ${varianza_salary:,.2f}')
desviacion_estandar_salary = df['SALARY'].std()
print(f'La desviación estándar de los salarios entre profesiones es de: ${desviacion_estandar_salary:,.2f}')
salary_max = df['SALARY'].max()
salary_min = df['SALARY'].min()
rango_salary = salary_max - salary_min
print(f'El rango de los salarios de las profesiones es: ${rango_salary:,.2f}')

La varianza de los salarios entre profesiones es: $1,359,909,953.44
La desviación estándar de los salarios entre profesiones es de: $36,876.96
El rango de los salarios de las profesiones es: $348,111.00


La varianza de los datos es alta, lo que nos indica que hay una dispercion entre los datos, adicional la desviacion estandar nos indica que las diferencias entre los datos y la media es de ~36,000, finalmente el rango nos demuestra que la distancia entre salarios es bastante amplia, lo que apoya a las medidas de dispersion.

In [ ]:
# Calcular la media truncada
media_salary = df['SALARY'].mean()
print(f'La media de los salarios es: ${media_salary:.2f}')

La media de los salarios es: $58136.68


In [ ]:
from scipy import stats
import pandas as pd
salary_sin_nan = df['SALARY'].dropna()
# Calcular la media truncada
media_truncada_salary = stats.trim_mean(salary_sin_nan, 0.05)
print(f'La media truncada de los salarios es: ${media_truncada_salary:.2f}')

La media truncada de los salarios es: $51631.97


Hay una diferencia de ~$6,000, lo que es un valor considerable por lo que podemos deducir que existen algunos outliers afectando la media original.

#Reto 3
Instrucciones
📂 Carga el dataset: Descarga el archivo Student_Performance.csv y súbelo a tu Google Drive. Luego, carga el dataset en un DataFrame de Pandas.

📊 Calcular los cuartiles: Calcula el primer cuartil (Q1), el segundo cuartil (Q2) o mediana, y el tercer cuartil (Q3) de la columna StudyTimeWeekly.

📐 Calcular percentiles: Calcula el percentil 25, 40 y 90 de la columna StudyTimeWeekly.

🔍 Análisis de resultados: Compara los resultados obtenidos e interpreta qué información nos proporcionan sobre la distribución del tiempo de estudio semanal en el dataset.

✅ Desafío adicional: Calcula el rango intercuartílico (IQR) y determina si existen valores atípicos en la columna StudyTimeWeekly.

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/DataScienceBedu/Datasets/S01/Reto_03_Student_Performance.csv')
# Calcular los cuartiles de la tarifa
Q1_StudyTimeWeekly = df['StudyTimeWeekly'].quantile(0.25)
Q2_StudyTimeWeekly = df['StudyTimeWeekly'].quantile(0.50)
Q3_StudyTimeWeekly = df['StudyTimeWeekly'].quantile(0.75)
print(f'Primer cuartil (Q1) del tiempo de estudio semanal: {Q1_StudyTimeWeekly:.2f} horas')
print(f'Segundo cuartil (Q2) del tiempo de estudio semanal (mediana): {Q2_StudyTimeWeekly:.2f} horas')
print(f'Tercer cuartil (Q3) del tiempo de estudio semanal: {Q3_StudyTimeWeekly:.2f} horas')
# Eliminar valores NaN en la columna 'Age'
StudyTimeWeekly_sin_nan = df['StudyTimeWeekly'].dropna()
percentil_25_StudyTimeWeekly = StudyTimeWeekly_sin_nan.quantile(0.25)
percentil_40_StudyTimeWeekly = StudyTimeWeekly_sin_nan.quantile(0.40)
percentil_90_StudyTimeWeekly = StudyTimeWeekly_sin_nan.quantile(0.90)
print(f'El percentil 25 del tiempo de estudio semanal es: {percentil_25_StudyTimeWeekly:.2f} horas')
print(f'El percentil 40 del tiempo de estudio semanal es: {percentil_40_StudyTimeWeekly:.2f} horas')
print(f'El percentil 90 del tiempo de estudio semanal es: {percentil_90_StudyTimeWeekly:.2f} horas')


Primer cuartil (Q1) del tiempo de estudio semanal: 5.04 horas
Segundo cuartil (Q2) del tiempo de estudio semanal (mediana): 9.71 horas
Tercer cuartil (Q3) del tiempo de estudio semanal: 14.41 horas
El percentil 25 del tiempo de estudio semanal es: 5.04 horas
El percentil 40 del tiempo de estudio semanal es: 7.94 horas
El percentil 90 del tiempo de estudio semanal es: 17.79 horas


La mayoría de los estudiantes estudian entre 5 y 14 horas semanales. La mediana de 9.71 h indica que el tiempo de estudio típico está cerca de las 10 horas por semana. Solo un pequeño grupo (10%) estudia más de 17.79 horas, lo cual puede representar a los estudiantes más comprometidos, exigidos o con más tiempo disponible.

In [ ]:
IQR = Q3_StudyTimeWeekly - Q1_StudyTimeWeekly

limite_inferior = Q1_StudyTimeWeekly - 1.5 * IQR
limite_superior = Q3_StudyTimeWeekly + 1.5 * IQR

# Filtrar outliers altos
outliers = df[df['StudyTimeWeekly'] > limite_superior]

print(f'IQR: {IQR:.2f} horas')
print(f'Límite superior: {limite_superior:.2f} horas')
print(f'Número de valores atípicos altos: {len(outliers)}')


IQR: 9.37 horas
Límite superior: 28.46 horas
Número de valores atípicos altos: 0
